src/app/clusters/commodity-metering-server/commodity-metering-server.h

```diff
@@ -0,0 +1,75 @@
+/**
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <stddef.h>
+
+#include <lib/core/Optional.h>
+
+#include <app-common/zap-generated/cluster-enums.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/util/basic-types.h>
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace CommodityMetering {
+
+class Instance : public AttributeAccessInterface
+{
+public:
+    Instance(const EndpointId & aEndpointId) : AttributeAccessInterface(MakeOptional(aEndpointId), Id), mEndpointId(aEndpointId) {}
+    ~Instance() override;
+
+    CHIP_ERROR Init();
+    void Shutdown();
+
+    // Attribute Accessors
+    const DataModel::Nullable<DataModel::List<Structs::MeteredQuantityStruct::Type>> & GetMeteredQuantity() const
```

Comment:
Why is this public API?  If it should be... It needs to document something about validity lifetimes of these references and the underlying buffers and whatnot.